sim/priest/talents.go

```diff
@@ -445,18 +402,15 @@ func (priest *Priest) applyImprovedDevouringPlague() {
 		ActionID:                 core.ActionID{SpellID: 63675},
 		SpellSchool:              core.SpellSchoolShadow,
 		ProcMask:                 core.ProcMaskProc,
-		Flags:                    core.SpellFlagIgnoreAttackerModifiers | core.SpellFlagIgnoreTargetModifiers | core.SpellFlagNoSpellMods,
+		Flags:                    core.SpellFlagIgnoreAttackerModifiers | core.SpellFlagNoSpellMods,
 		DamageMultiplier:         1,
 		DamageMultiplierAdditive: 1,
 		ThreatMultiplier:         1,
 		CritMultiplier:           priest.DefaultSpellCritMultiplier(),
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-			dp := priest.DevouringPlague
-			dot := dp.Dot(target)
-
-			// No need to use snapshot dmg. It won't be initialized in time and we're in the same sim cycle
-			// so all mods and buffs will be the same
-			dmg := float64(dot.NumberOfTicks*int32(dp.ExpectedTickDamage(sim, target))*priest.Talents.ImprovedDevouringPlague) * 0.15
+			dot := priest.DevouringPlague.Dot(target)
+			dpTickDamage := dot.CalcSnapshotDamage(sim, target, dot.OutcomeExpectedMagicSnapshotCrit)
```

Comment:
`dpTickDamage := dot.SnapshotBaseDamage` it should be already calculated. Also using the `OutcomeExpectedMagicSnapshotCrit` will bake crit inside the base damage then you apply another outcome that can crit as well making this double dip on crit which seems wrong.

Just using the dots snapshotted base damage should give you exactly that the expected non-crit damage per tick.